6_visualize/src/prep_precip_fun.R

```diff
@@ -0,0 +1,34 @@
+
+prep_precip_fun <- function(precip_rasters, precip_bins, timestep){
+
+  breaks <- unique(c(precip_bins$left_break,
+                     precip_bins$right_break))
+  breaks[which(breaks == Inf)] <- 100000 # Just quiets an error
+  breaks <- c(0, 1, breaks[2:length(breaks)]) # Add a tiny break for alpha
+
+  rgb_ramp <- col2rgb(precip_bins$col, alpha = TRUE)
+  rgb_ramp <- cbind(rgb_ramp[,1],rgb_ramp)
+  rgb_ramp["alpha", 1:4] <- c(0, 100, 180, 220) # fade in from 0
+  rgb_ramp["alpha", ] <- rgb_ramp["alpha", ] * 0.8 # decrease all alpha a bit
```

Comment:
Thought it would be nice to have 0 fade out -- just an idea. I think it looks good and helps hide the edge of the precip data field.

Reply:
@jread-usgs had proposed that we should avoid transparency in order to stick to a 256-color palette for gif, but I've gotta say I really like how these transparent points interact with rivers and state boundaries.

Reply:
I guess I though that transparency within layers that go to PNG would work but transparency of the PNG it's self wouldn't?

Reply:
So -- it _does_ work, and I'm predicting that @jread-usgs will be on board with this. But the introduction of transparency means we can no longer guarantee that the png will have <= 256 colors, which means we have less control over the conversion to 256 colors in the gif, which is why I want jordan to weigh in.

Reply:
It'll work, and I agree that the look is definitely :+1: Nice job!

I would suggest we alter it though and put transparency directly into the color param in the yaml, that way it will show up the same way in the legend as on the map. We can address that later, but that would be my suggestion. 

and yes, with this approach, we'll need to put a bit more trust that the png -> gif conversion is going to keep colors consistent across frames.